File: examples/python/control_flow.py
# Control flow in Python

# If statement
value = 10
if value > 0:
    print("Positive")
elif value < 0:
    print("Negative")
else:
    print("Zero")

# Match statement (Python 3.10+, similar to switch)
status = "success"
match status:
    case "success":
        print("Operation succeeded")
    case "error":
        print("Operation failed")
    case _:
        print("Unknown status")

# For loop
for i in range(5):
    print(f"Iteration {i}")

# While loop
count = 0
while count < 3:
    print(f"Count: {count}")
    count += 1

File: examples/rust/control_flow.rs
// Control flow in Rust

fn main() {
    // If statement
    let value = 10;
    if value > 0 {
        println!("Positive");
    } else if value < 0 {
        println!("Negative");
    } else {
        println!("Zero");
    }

    // Match statement (similar to switch)
    let status = "success";
    match status {
        "success" => println!("Operation succeeded"),
        "error" => println!("Operation failed"),
        _ => println!("Unknown status"),
    }

    // For loop
    for i in 0..5 {
        println!("Iteration {}", i);
    }

    // While loop
    let mut count = 0;
    while count < 3 {
        println!("Count: {}", count);
        count += 1;
    }
}
